cura/Settings/MachineManager.py

```diff
@@ -1649,3 +1657,47 @@ def getAbbreviatedMachineName(self, machine_type_name: str) -> str:
                 abbr_machine += stripped_word
 
         return abbr_machine
+
+    def getMachineTypeNameFromId(self, machine_type_id: str) -> str:
+        machine_type_name = ""
+        results = self._container_registry.findDefinitionContainersMetadata(id = machine_type_id)
+        if results:
+            machine_type_name = results[0]["name"]
+        return machine_type_name
+
+    @pyqtSlot(QObject)
+    def associateActiveMachineWithPrinterDevice(self, printer_device: Optional["PrinterOutputDevice"]) -> None:
+        if not printer_device:
+            return
+
+        Logger.log("d", "Attempting to set the network key of the active machine to %s", printer_device.key)
+
+        global_stack = self._global_container_stack
+        if not global_stack:
+            return
+
+        meta_data = global_stack.getMetaData()
+
+        if "um_network_key" in meta_data:  # Global stack already had a connection, but it's changed.
```

Comment:
It changed? Where did you get that from?

Reply:
I probably copy-and-pasted this from somewhere. I've updated the comments in da351da61. This comment meant to say that this `if`-block handles the case in which the global stack previously had a connection already.